core/domain/exp_services.py

```diff
@@ -2053,6 +2053,245 @@ def get_interaction_id_for_state(exp_id, state_name):
         'There exist no state in the exploration with the given state name.')
 
 
+def _get_checkpoints_in_order(init_state_name, states):
```

Comment:
Why not use the already created function of '_get_checkpoints_in_order'? I think this is the same repeated code.

Reply:
Done.